### PR TITLE
Better help messages when configuration won't work

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: wrapper/wrapper [hunk 1]
```diff
@@ -14,11 +14,23 @@ if [ -v MIR_SERVER_WAYLAND_HOST ]; then
...
   ln -sf "${host_wayland}.lock" "$XDG_RUNTIME_DIR"
+elif [ -v DISPLAY ] && ! snapctl is-connected x11; then
+  echo "ERROR: DISPLAY [${DISPLAY}] set, but you need to connect the x11 interface"
+  echo "Try: sudo snap connect" $SNAP_NAME":x11"
+  exit 1
+elif [ $(id -u) != 0 ] && ! [ -v DISPLAY ] && ! snapctl is-connected login-session-control; then
+  echo "ERROR: You need one of:"
+  echo "  1. An X11 host specified by DISPLAY environment variable"
+  echo "  2. The login-session-control interface connected"
+  echo "     Try: sudo snap connect" $SNAP_NAME":login-session-control"
```
Comment (Saviq) [inline wrapper/wrapper]: Why the quotes around `$SNAP_NAME`?

Comment (AlanGriffiths) [reply wrapper/wrapper]: Inherited by cut & paste

File: wrapper/wrapper [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!/bin/bash
 set -e
 
-if [ $(id -u) != 0 ] && ! [ -v DISPLAY ] && ! snapctl is-connected login-session-control; then
-  echo "ERROR: You need one of: root, a DISPLAY, or to connect the login-session-control interface."
-  echo "e.g. sudo snap connect" $SNAP_NAME":login-session-control :login-session-control"
+if ! snapctl is-connected graphics-core20; then
+  echo "ERROR: You need to connect the graphics-core20 interface."
+  echo "e.g. sudo snap connect $SNAP_NAME:graphics-core20 mesa-core20:graphics-core20"
```
Comment (Saviq) [inline wrapper/wrapper:6]: Slightly simpler:
```suggestion
  echo "e.g. sudo snap connect $SNAP_NAME:graphics-core20 mesa-core20"
```

File: wrapper/wrapper [hunk 3]
```diff
@@ -14,11 +14,23 @@ if [ -v MIR_SERVER_WAYLAND_HOST ]; then
...
   ln -sf "${host_wayland}.lock" "$XDG_RUNTIME_DIR"
+elif [ -v DISPLAY ] && ! snapctl is-connected x11; then
+  echo "ERROR: DISPLAY [${DISPLAY}] set, but you need to connect the x11 interface"
+  echo "Try: sudo snap connect $SNAP_NAME:x11"
+  exit 1
+elif [ $(id -u) != 0 ] && ! [ -v DISPLAY ] && ! snapctl is-connected login-session-control; then
+  echo "ERROR: You need one of:"
+  echo "  1. An X11 host specified by DISPLAY environment variable"
+  echo "  2. The login-session-control interface connected"
+  echo "     Try: sudo snap connect $SNAP_NAME:login-session-control"
```
Comment (Saviq) [inline wrapper/wrapper:30]: When I do that, it failed with "operation not permitted" and/or "Permission denied", even if I used `--console-provider=logind`. Something I'm missing? Anything more we should say here?

Comment (AlanGriffiths) [reply wrapper/wrapper:30]: Logind can still refuse, e.g. if DRM master on the VT has been granted to Mutter.

Comment (Saviq) [reply wrapper/wrapper:30]: So I should log out of my session / switch to GDM for it to work?

Comment (AlanGriffiths) [reply wrapper/wrapper:30]: Just switch to a different VT and login?

**Review by Saviq — APPROVED** [review]
Yup.

bors merge

**Pull request merged**